Automattic/Gravatar-SDK-Android
# QuickEditor: Take photo action in the media picker

Closes #225

### Description

This PR implements the `Take photo` option from the media picker popup.

An important bit here is the `FileProvider` config - this needs to be custom so that it will work when the app implementing our library has its own `FileProvider` setup.  You can read about it [here](https://commonsware.com/blog/2017/06/27/fileprovider-libraries.html). 
I've also added a config to our DemoApp to make sure this works (that's how I discovered this needs to be handled).


https://github.com/user-attachments/assets/7e9e399d-957e-46a6-a64a-cf6d0e70e0d9



### Testing Steps
**Important Note**: The backend is currently returning the wrong format for the `updated date` field. As a workaround, you can apply this patch to comment that field from the Avatar object.
[Gravatar-Android-13-04-48.patch](https://github.com/user-attachments/files/16455284/Gravatar-Android-13-04-48.patch)

1. Make sure you have at least two avatars uploaded on Gravatar
2. Run the DemoApp 
3. Go to Avatar Update tab
4. Tap `Update Avatar` button
5. Follow the steps to log in (if necessary)
8. Tap the `Upload Image` button
9. Tap the `Take photo`
10. Take photo with camera
11. Crop if you want and confirm
12. Confirm you see the photo in the list with the progress indicator
13. Confirm the photo was uploaded and can be selected
